tensorboard/plugins/profile/tf_op_profile/tf-op-table.html

```diff
@@ -0,0 +1,209 @@
+
+<!--
+  tf-op-table-entry is a row in a tf-op-table.
+  It describes an op, or group of ops (the node property).
+  Its background contains a bar, whose size shows the time taken, and whose
+  color shows the op's utilization.
+  If the node has children, the tf-op-table-entry has child entries, and can be
+  expanded/collapsed.
+-->
+<dom-module id="tf-op-table-entry">
+<template>
+<style include="tf-op-table-styles">
+#row {
+  position: relative;
+  box-shadow: inset 0 1px 0 0 rgba(0,0,0,0.06);
+  z-index: 0;
+}
+#bar {
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  z-index: -1;
+}
+:host {
+  display: block;
+}
+:host.selected {
+  background-color: rgba(0,0,0,0.04);
+}
+:host.selected > #row {
+  font-weight: bold;
+}
+#row:hover {
+  background-color: rgba(0,0,0,0.05);
+}
+#disclosure {
+  display: inline-block;
+  width: 1em;
+}
+#name {
+  font-family: monospace;
+}
+</style>
+<div id="row"
+     on-click="_handleHeaderClick"
+     on-mouseenter="_handleHeaderMouseEnter"
+     on-mouseleave="_handleHeaderMouseLeave"
+     hidden="[[!level]]">
+  <div id="bar"></div>
+  <span id="time">{{_percent(node.metrics.time)}}</span>
+  <span id="name" style$="padding-left:[[level]]em;">
+    <span id="disclosure">
+      <span hidden="[[!node.children.length]]">
+        <span hidden="[[expanded]]">&#x25b6;</span>
+        <span hidden="[[!expanded]]">&#x25bc;</span>
+      </span>
+    </span>{{node.name}}
+  </span>
+  <span id="provenance">{{_provenance(node)}}</span>
+</div>
+<template is="dom-if" if="[[expanded]]">
+  <template is="dom-repeat" items="{{node.children}}" sort="_sortEntry">
+    <tf-op-table-entry
+      node="[[item]]"
+      level={{_nextLevel(level)}}
+      table={{table}}>
+    </tf-op-table-entry>
+  </template>
+</template>
+</dom-module>
+
+<script>
+import {flameColor, percent, utilization} from './utils.js';
+
+Polymer({
+  is: 'tf-op-table-entry',
+  properties: {
+    node: {
+      type: Object,
+    },
+    level: {
+      type: Number,
+      value: 0,
+    },
+    table: {
+      type: Object,
+    },
+    expanded: {
+      type: Boolean,
+      value: false,
+      notify: true,
+    },
+  },
+  _eq: function(x, y) {
+    return x == y;
+  },
+  _sortEntry: function(left, right) {
+    var leftTime = left.metrics ? left.metrics.time : 0;
+    var rightTime = right.metrics ? right.metrics.time : 0;
+    return rightTime - leftTime;
+  },
+  ready: function() {
+    if (!this.node || !this.node.metrics || !this.$.bar) return;
+    this.$.bar.style.width = percent(this.node.metrics.time);
+    this.$.bar.style.backgroundColor = flameColor(utilization(this.node), 1, 0.2);
+  },
+  _nextLevel: function(level) { return level + 1; },
+  _handleHeaderClick: function(e) {
+    this.expanded ^= true;
+    if (this.table._selectedEntry) this.table._selectedEntry.classList.remove('selected');
+    this.classList.add('selected');
+    this.table._selectedEntry = this;
+    this.table.selected = this.node;
```

Comment:
Likewise, make `selected` a property of `tf-op-table-entry` and upward bind, so you can call `this.set('selected', this.node);`.